services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -179,13 +222,16 @@ async def remove_disconnected_user_resources(
                     resource_value,
                     dead_key,
                 )
-                # inform that the project can be closed on the backend side
-                await emit(
-                    event="SIGNAL_PROJECT_CLOSE",
-                    user_id=None,
-                    project_uuid=resource_value,
-                    app=app,
-                )
+
+                # --- actual cleanup happens now ---
+                if resource_name == "project_id":
+                    # inform that the project can be closed on the backend side
+                    await emit(
+                        event="SIGNAL_PROJECT_CLOSE",
+                        user_id=None,
+                        project_uuid=resource_value,
+                        app=app,
+                    )
```

Comment:
This should be done before removing the resource from redis (line 217 above) -  that might be the issue you are refering to @odeimaiz ?

Reply:
Right, before removing the project_id resource from the registry, we need to make sure that its services were successfully stopped.

Reply:
It would also be nice to know what the status of that resource is. In case of a study if it's open or being closed.

Reply:
> Right, before removing the project_id resource from the registry, we need to make sure that its services were successfully stopped.

exactly

Reply:
Done